python/ray/autoscaler/_private/aws/config.py

```diff
@@ -415,7 +415,13 @@ def _configure_subnet(config):
                 "to populate the list of subnets and trying this again.".
                 format(config["provider"]["availability_zone"]))
 
-    subnet_ids = [s.subnet_id for s in subnets]
+    # Use subnets in only one VPC. This is to make sure we don't attempt
+    # to launch a node into a VPC outside the security group
+    # generated by _configure_security_group.
```

Comment:
This comment is a bit misleading. I think it should say something more along the lines of: "This is because we do not want to create security groups for every VPC the user has." 
^ My understanding is that this is the real fundamental reason here.